src/gui/DatabaseWidget.cpp

```diff
@@ -1220,3 +1227,15 @@ void DatabaseWidget::closeUnlockDialog()
 {
     m_unlockDatabaseDialog->close();
 }
+
+void DatabaseWidget::showMessage(const QString& text, MessageWidget::MessageType type)
+{
+    m_messageWidget->showMessage(text, type);
+}
+
+void DatabaseWidget::hideMessage()
+{
+    if (m_messageWidget->isVisible()) {
```

Comment:
Why not add the `isVisible()` condition in `MessageWidget::hide` and call `m_messageWidget->hide()` directly? This code is also repeated in `EditWidget.cpp`